src/main/java/shadows/apotheosis/adventure/affix/socket/SocketingRecipe.java

```diff
@@ -41,21 +41,22 @@ public boolean matches(Container pInv, Level pLevel) {
 	 * Returns an Item that is the result of this recipe
 	 */
 	@Override
-	public ItemStack assemble(Container pInv) {
-		ItemStack out = pInv.getItem(0).copy();
-		if (out.isEmpty()) return ItemStack.EMPTY;
-		out.setCount(1);
-		int sockets = SocketHelper.getSockets(out);
-		List<ItemStack> gems = SocketHelper.getGems(out, sockets);
-		for (int idx = 0; idx < gems.size(); idx++) {
-			Gem gem = GemItem.getGem(gems.get(idx));
-			if (gem == null) {
-				gems.set(idx, pInv.getItem(1));
-				break;
-			}
-		}
-		SocketHelper.setGems(out, gems);
-		return out;
+	public ItemStack assemble(Container inventory) {
+		var result = inventory.getItem(0).copy();
+		if (result.isEmpty()) return ItemStack.EMPTY;
+		result.setCount(1);
+		var sockets = SocketHelper.getSockets(result);
+		var gems = SocketHelper.getGems(result, sockets);
+		var gemStack = inventory.getItem(1).copy();
+		var event = new ItemSocketingEvent(result, gemStack);
```

Comment:
Just for reference, this is the only acceptable use of `var` in this entire method.